src/Platform/Microsoft.Testing.Platform/Helpers/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+
+namespace Microsoft.Testing.Platform.Helpers;
+
+internal static class IEnumerableExtensions
+{
+    public static IReadOnlyCollection<T> ToReadOnlyCollectionEnumerator<T>(this IEnumerable<T> enumerable)
```

Comment:
The method suggests that it is returning an enumerator but that's not the case. I would prefer `ToReadOnlyCollection`.